src/layout.hpp

```diff
@@ -17,28 +17,33 @@
 
 #include <string>
 #include <unordered_map>
-#include <vector>
+
+#include "status.hpp"
 
 namespace ovms {
 
-enum Mode { FIXED,
-    AUTO };
-using shape_t = std::vector<size_t>;
+class LayoutConfiguration {
+    std::string tensor;
+    std::string model;
+
+public:
+    LayoutConfiguration() = default;
+    LayoutConfiguration(const char* Layout);
+    LayoutConfiguration(const std::string& Layout);
+    LayoutConfiguration(const std::string& tensorLayout, const std::string& modelLayout);
 
-struct ShapeInfo {
-    Mode shapeMode = FIXED;
-    shape_t shape;
+    const std::string& getTensorLayout() const;
```

Comment:
Maybe add utility functions as:
size_t LayoutConfiguration.getDimensionPos("N")?

Do we plan to keep dimension names here as well in the end?

Reply:
Add task to document batch size position other than 0th in DAG. This should not be allowed since dags demultiplication always assumes batch is on first dimension.

Reply:
Add task to implement gathering layout hint information from Model Optimizer

Reply:
Add warning when using shape/bs=auto to use dynamic shape

Reply:
Make full network reload if layout configuration differs.
This would be the workaround for: https://jira.devtools.intel.com/browse/CVS-74065